catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -156,9 +156,17 @@ public MetacardImpl(Metacard metacard, MetacardType type) {
             throw new IllegalArgumentException(
                     MetacardType.class.getName() + " instance should not be null.");
         }
+
+        if (metacard.getSourceId() != null) {
```

Comment:
It looks like this should be added to `MetacardImpl(Metacard metacard)` as well so both constructors are consistent.